### PR TITLE
[FLINK-16966][metrics][infuxdb] Convert InfluxDB reporter to plugin

### PR DESCRIPTION
**Pull request by complone**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/monitoring/metrics.md [hunk 1]
```diff
@@ -663,6 +663,7 @@ Example configuration:
 
 {% highlight yaml %}
 
+metrics.reporter.influxdb.factory.class: org.apache.flink.metrics.influxdb.InfluxdbReporterFactory
 metrics.reporter.influxdb.class: org.apache.flink.metrics.influxdb.InfluxdbReporter
```
Comment (zentol) [inline docs/monitoring/metrics.md:667]: We should only advertise the plugin approach.

```suggestion
```

File: docs/monitoring/metrics.zh.md [hunk 2]
```diff
@@ -663,6 +663,7 @@ Example configuration:
 
 {% highlight yaml %}
 
+metrics.reporter.influxdb.factory.class: org.apache.flink.metrics.influxdb.InfluxdbReporterFactory
 metrics.reporter.influxdb.class: org.apache.flink.metrics.influxdb.InfluxdbReporter
```
Comment (zentol) [inline docs/monitoring/metrics.zh.md:667]: We should only advertise the plugin approach.

```suggestion
```

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 3]
```diff
@@ -591,6 +591,13 @@ function setup_flink_slf4j_metric_reporter() {
   set_config_key "metrics.reporter.slf4j.interval" "${INTERVAL}"
 }
 
+function setup_flink_influxdb_metric_reporter() {
```
Comment (zentol) [inline flink-end-to-end-tests/test-scripts/common.sh:594]: without a test to make use of this there is little reason to add this method,

**Comment by zentol** [comment]
The file permission change was likely due to git, see this [stackoverflow](https://stackoverflow.com/questions/1580596/how-do-i-make-git-ignore-file-mode-chmod-changes).

With these changes reverted and the minor comments addressed this would be good to go 👍

**Comment by complone** [comment]
> 文件权限更改可能是由于git引起的，请参阅此[stackoverflow](https://stackoverflow.com/questions/1580596/how-do-i-make-git-ignore-file-mode-chmod-changes)帖子。
> 
> 恢复了这些更改，并解决了一些小意见，这将是一个很好的选择 👍

OK, I'll correct it

**Comment by zentol** [comment]
@complone ping

**Comment by complone** [comment]
> @complone ping

Please give me some time. I will resubmit PR tomorrow

**Comment by zentol** [comment]
@complone Sure, no problem; the 1.11 feature freeze is on Friday so if we get it in before then we're good :)

**Comment by complone** [comment]
> @complone当然，没问题；1.11功能冻结是在星期五，所以如果我们早点安装它，那就好了:)
Hello, I have modified and resubmitted a new pull request. Maybe I can close the current pull request
https://github.com/apache/flink/pull/12153

**Comment by zentol** [comment]
Subsumed by #12153.

**Pull request closed without merging**